### PR TITLE
23851 adobe fonts

### PR DESCRIPTION
**Pull request by CristianSpatari**

### CODE REVIEW

File: public/editor-client/src/fonts/index.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  mockTypeKitData.kit.families.forEach((family) => {
+    fonts.kit.families.push({
+      id: family.id,
+      family: family.name,
+      category: family.slug,
+      kind: "webfonts#webfont",
+      subsets: [family.css_names[0]],
+      variants: family.variations
+    });
+  });
```
Comment (victor-pg) [inline public/editor-client/src/fonts/index.ts:55]: use `array.map` for this
```ts
const families = mockTypeKitData.kit.families.map(()=>(
    {
      id: family.id,
      family: family.name,
      category: family.slug,
      kind: "webfonts#webfont",
      subsets: [family.css_names[0]],
      variants: family.variations
    }
  )
)

return {
    kit: {
      id: mockTypeKitData.kit.id,
      families
    }
  }
```

File: brizy.php
Comment (victor-pg) [inline brizy.php:1]: rollback

File: public/editor-client/src/fonts/index.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  subsets: string[];
+  variants: string[];
+}
+
+interface Fonts {
+  kit: {
+    id: string;
+    families: Font[];
+  };
+}
```
Comment (victor-pg) [inline public/editor-client/src/fonts/index.ts:36]: make `types.ts` file and move there

File: public/editor-client/src/fonts/index.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+function convertDataToLocal(mockTypeKitData: KitData): Fonts {
```
Comment (victor-pg) [inline public/editor-client/src/fonts/index.ts:38]: make arrow function

File: public/editor-src/editor/js/redux/actions2.ts [hunk 4]
```diff
@@ -28,6 +28,7 @@ type StrictFonts = Required<RFonts>;
 export type FontKeyTypes = keyof StrictFonts;
 
 type FontPayload<T extends FontKeyTypes> = {
+  id?: string
```
Comment (victor-pg) [inline public/editor-src/editor/js/redux/actions2.ts:31]: review `id`

**Review by victor-pg — CHANGES_REQUESTED** [review]

File: brizy.php
Comment (victor-pg) [inline brizy.php:1]: rollback

File: public/editor-client/src/fonts/index.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+//     const config = getConfig();
+//
+//     if (!config) {
+//       throw new Error(t("Invalid __BRZ_PLUGIN_ENV__"));
+//     }
+//
+//     const { editorVersion, url: _url, hash, actions } = config;
+//
+//     const url = makeUrl(_url, {
+//       hash,
```
Comment (victor-pg) [inline public/editor-client/src/fonts/index.ts:56]: why is commented ?

**Review by victor-pg — CHANGES_REQUESTED** [review]

File: public/editor-client/src/api/index.ts [hunk 6]
```diff
@@ -907,6 +907,67 @@ export const updateBlockScreenshot = async ({
 
 //#endregion
 
+//#region Screenshots
```
Comment (maxval1) [inline public/editor-client/src/api/index.ts:910]: rename region to AdobeFonts

File: public/editor-client/src/api/index.ts [hunk 7]
```diff
@@ -907,6 +907,67 @@ export const updateBlockScreenshot = async ({
...
+  }
+  const { url: _url, hash, editorVersion, actions } = config;
+
+  const url = makeUrl(_url, {
+    hash,
+    action: actions.addAccount,
+    version: editorVersion
+  });
+
+  return request(url, {
```
Comment (maxval1) [inline public/editor-client/src/api/index.ts:960]: transform to `async/await`

File: public/editor-client/src/config.ts [hunk 8]
```diff
@@ -144,17 +146,23 @@ const apiReader = parseStrict<PLUGIN_ENV["api"], API>({
       Obj.readKey("imagePatterns"),
       Obj.read,
       parseStrict<PLUGIN_ENV, ImagePatterns>({
-        full: pipe(
-          mPipe(Obj.readKey("full"), Str.read),
-          throwOnNullish("Invalid API: ImagePatterns full pattern")
+        full: optional(
```
Comment (maxval1) [inline public/editor-client/src/config.ts:149]: optional?

**Review by maxval1 — CHANGES_REQUESTED** [review]

**Pull request merged**